src/GeometricalAnalysisTools.cpp

```diff
@@ -997,6 +998,164 @@ GeometricalAnalysisTools::ErrorCode GeometricalAnalysisTools::DetectSphereRobust
 	return NoError;
 }
 
+bool GeometricalAnalysisTools::Landau_Smith(std::vector<CCVector3> xy, CCVector3& center, float& radius) {
```

Comment:
I wanted to avoid extra conversion between CCVector3 and CCVector2. But with your major update of the circle projection algorithm, it becomes more natural to have CCVector2 here now